lib/uksched/sched.c

```diff
@@ -302,6 +302,10 @@ void uk_sched_thread_terminate(struct uk_thread *thread)
 	uk_pr_debug("%p: thread %p (%s) terminated\n",
 		    sched, thread, thread->name ? thread->name : "<unnamed>");
 
+	if (uk_thread_in_waitq(thread))
+		uk_pr_warn("%p: thread %p (%s) terminated while waiting\n",
```

Comment:
Does it really have to be a warning level message? I would expect a debug/info level message maybe.